testing/test_warnings.py

```diff
@@ -678,7 +678,7 @@ def test_issue4445_preparse(self, testdir):
         # thrown by an errorneous conftest.py
         result.stdout.fnmatch_lines_random(
             [
-                "*config{sep}__init__.py:915*".format(sep=os.sep),
+                "*config{sep}__init__.py*".format(sep=os.sep),
```

Comment:
`:*` ?

Reply:
what exactly do you mean with that? Should the colon be again part of the string to check?

Reply:
For that specific file (config/__init__.py) is was wondering if checking again against line numbers is maybe not such a bad idea. For more unique files to check for a filename is ok i think, but __init__.py is so generic and has lots of code inside that stacklevel changes that point to another location in that file would be eaten up by this check...

Reply:
Yeah, true.
I think we should find a way to match the function name - maybe through reconfiguring display of warnings, if necessary?

Reply:
> How to do that? Using `inspect.stack` and choose the second `FrameInfo` object of the returned list?

Give it a try I guess.  Otherwise I've thought about changing the format of warnings to include the function name.
Maybe @nicoddemus can help?

Reply:
I added a new commit that uses inspect.stack